actions/action_emitter_mixin.dart/change_notifier_action_emitter_mixin.dart

```diff
@@ -0,0 +1,17 @@
+mixin ChangeNotifierActionEmitterMixin<T> on ChangeNotifier implements ActionEmitter<T> {
```

Comment:
Is it possible to create mixin for providers that plain classes that don't extend anything?

```dart
final myPresenter = Provider((ref) => MyPresenter());

class MyPresenter {
   ...
}
```

Reply:
It could be done but there might be some problems (see the attached code snippet).

I would argue it is better to not have a mixin for plain classes. The user can always manually implement the `ActionEmitter` interface.

```dart
mixin ActionEmitterMixin<T> implements ActionEmitter<T> {
  final _actionSubject = PublishSubject<T>();

  @override
  Stream<T> get action => _actionSubject;

  void dispose() {
    // Having a dispose() method would be risky, in cases where the host class also has a dispose method.
    // Problem is that we cannot call super.dispose() here.
    _actionSubject.close();
  }

  void disposeAction(){
    // This would work, but the user would have to remember to call this method.
    _actionSubject.close();
  }

  @protected
  void emitAction(T action) {
    _actionSubject.add(action);
  }
}
```

Reply:
Aha, okay 👍